public/client/TracyProfiler.cpp

```diff
@@ -4291,9 +4291,16 @@ TRACY_API TracyCZoneCtx ___tracy_emit_zone_begin_callstack( const struct ___trac
         TracyQueueCommitC( zoneValidationThread );
     }
 #endif
-    tracy::GetProfiler().SendCallstack( depth );
-    {
-        TracyQueuePrepareC( tracy::QueueType::ZoneBeginCallstack );
+    if (depth > 0 && tracy::has_callstack()) {
+        tracy::GetProfiler().SendCallstack( depth );
+        {
+            TracyQueuePrepareC( tracy::QueueType::ZoneBeginCallstack );
+            tracy::MemWrite( &item->zoneBegin.time, tracy::Profiler::GetTime() );
+            tracy::MemWrite( &item->zoneBegin.srcloc, (uint64_t)srcloc );
+            TracyQueueCommitC( zoneBeginThread );
+        }
+    } else {
+        TracyQueuePrepareC( tracy::QueueType::ZoneBegin );
         tracy::MemWrite( &item->zoneBegin.time, tracy::Profiler::GetTime() );
         tracy::MemWrite( &item->zoneBegin.srcloc, (uint64_t)srcloc );
         TracyQueueCommitC( zoneBeginThread );
```

Comment:
There is shared code between the branches, maybe it should be moved out of the condition?